packages/liveblocks-react/src/liveblocks.tsx

```diff
@@ -1158,54 +1174,131 @@ function useDeleteAiChat() {
+ *
+ * You can also pass the chat ID dynamically if it's not known when calling the hook.
+ *
+ * @example
+ * const sendAiMessage = useSendAiMessage();
+ * sendAiMessage({ chatId: "chat-id", text: "Hello, Liveblocks AI!" });
+ *
+ * @example
+ * const sendAiMessage = useSendAiMessage();
+ * sendAiMessage({ chatId: "chat-id", text: "Hello, Liveblocks AI!", copilotId: "co_xxx" });
+ */
+function useSendAiMessage(
+  chatId?: string,
+  options?: UseSendAiMessageOptions
+):
+  | ((message: string | SendAiMessageOptions) => AiUserMessage)
+  | ((message: WithRequired<SendAiMessageOptions, "chatId">) => AiUserMessage) {
   const client = useClient();
-  const copilotId = options?.copilotId;
 
   return useCallback(
-    (message: string) => {
+    (message: string | SendAiMessageOptions) => {
+      const {
+        text: messageText,
+        chatId: messageOptionsChatId,
+        ...messageOptions
+      } = typeof message === "string" ? { text: message } : message;
+      const resolvedChatId =
+        messageOptionsChatId ??
+        chatId ??
+        // The `useSendAiMessage` overloads prevent this scenario from happening
+        // at the type level, and this error prevents it from happening at runtime.
+        raise(
+          "chatId must be provided to either `useSendAiMessage` or its returned function."
+        );
+
       const messages = client[kInternal].ai.signals
-        .getChatMessagesForBranchΣ(chatId)
+        .getChatMessagesForBranchΣ(resolvedChatId)
         .get();
 
       const lastMessageId = messages[messages.length - 1]?.id ?? null;
 
-      const content = [{ type: "text" as const, text: message }];
+      const content = [{ type: "text" as const, text: messageText }];
       const newMessageId = client[kInternal].ai[
         kInternal
       ].context.messagesStore.createOptimistically(
-        chatId,
+        resolvedChatId,
         "user",
         lastMessageId,
         content
       );
+      const newMessage = client[kInternal].ai[
+        kInternal
+      ].context.messagesStore.getMessageById(newMessageId) as AiUserMessage;
 
       const targetMessageId = client[kInternal].ai[
         kInternal
       ].context.messagesStore.createOptimistically(
-        chatId,
+        resolvedChatId,
         "assistant",
         newMessageId
       );
 
       void client[kInternal].ai.askUserMessageInChat(
-        chatId,
+        resolvedChatId,
         { id: newMessageId, parentMessageId: lastMessageId, content },
         targetMessageId,
         {
-          stream: options?.stream,
-          copilotId: copilotId as CopilotId | undefined,
-          timeout: options?.timeout,
+          stream: messageOptions.stream ?? options?.stream,
+          copilotId: (messageOptions.copilotId ?? options?.copilotId) as
+            | CopilotId
+            | undefined,
+          timeout: messageOptions.timeout ?? options?.timeout,
         }
       );
+
+      return newMessage;
```

Comment:
I guess we are kind of required to return the message for the AiComposer PR. Also, this would match `useCreateComment`'s signature too, so it makes sense, right?